test/reline/test_face.rb

```diff
@@ -160,7 +160,15 @@ def test_invalid_rgb?
       assert_equal false, @config.send(:rgb_expression?, "#FFFFF")
     end
 
-    def test_format_to_sgr_preserves_order
+    def test_define_method_proxies_the_order_of_arguments
+      pend "skip because test-unit-rr could not be loaded" if ENV['TEST_UNIT_RR'] == 'no'
+      mock(@config).format_to_sgr(
```

Comment:
Looks like we're essentially testing if calling `to_a` on a double splatted argument `**values` would preserve the order of keys. This is the behaviour of Ruby itself. Do we need to test it in Reline?

If we really want to make sure we store the correct result, why don't we test it end-to-end?

```rb
def test_define_method_proxies_the_order_of_arguments
  @config.define :default, foreground: :blue, style: [:bold, :italicized], background: :red
  assert_equal(
    { foreground: :blue, style: [:bold, :italicized], background: :red, escape_sequence: "#{RESET_SGR}\e[34;1;3;41m" },
    @config.instance_variable_get(:@definition)[:default]
  )
end
```

Reply:
End to end test is already implicitly done by some tests like https://github.com/ruby/reline/blob/master/test/reline/test_face.rb#L70.
Here I want to test the internal behavior of the `#define` method explicitly to explain the intent of the implementation. This is because I use a mock

Reply:
> End to end test is already implicitly done by some tests like

Yes I'm aware of them 🙂 

> explicitly to explain the intent of the implementation

I'm not against covering an implementation with multiple levels of tests. For example, it's good to have tests for the private `format_to_sgr` method too.

I'm simply not sure if the coverage we get for this case is worth introducing a new dependency and a test setup specifically for it. It'd be different if we can find other places to utilise it too though.